### PR TITLE
Added tooltip for when a task is empty

### PR DESCRIPTION
**Pull request by TheHollidayInn**
This should address this issue: https://github.com/HabitRPG/habitrpg/issues/4692 .

screen shot 2015-03-23 at 7 34 50 pm

### CODE REVIEW

**Comment by Alys** [comment]
Oh nice! 

There's a slight problem with the text though. "taskType" will be "habit", "daily", "todo", or "reward", whereas the correct name for each should have an initial capital letter ("Habit"), and "todo" should be "To-Do". Also, the taskType string wouldn't be converted to the player's language.

Can I suggest that the text is just "Enter the task's title first" or something like that. It's generic enough to work for all four types of tasks. (FYI, in the task edit screen, we are going to be changing "Text" to "Title", so using title in the tooltip will be consistent.)

**Comment by TheHollidayInn** [comment]
Adjusted! Although, my text editor seems to have altered some spacing.. I thought I had fixed this issue.

**Comment by Alys** [comment]
I believe that removing spaces at the ends of lines is good. I personally don't think you should worry that your editor is doing that. When I pull changes down locally, I use `git diff -w` as well as `git diff` to examine them (-w ignores whitespace changes), so it doesn't bother me.

I'll try to get this reviewed and merged in the next couple of days, if no one else gets in first.

**Comment by crookedneighbor** [comment]
@TheHollidayInn Actually, your editor was removing extra white space, not adding it. So, that's a plus for me! I hate extra white space and have my editor configured to remove it automatically on save. 

If we wanted to, there is a way to make the task type appear in the tooltip. (Doing this from memory, so bear with me if I do something wrong:

### JSON

``` json
"emptyTask": "Enter the <%= taskType %> title first."
```

### Jade

``` jade
tooltip=env.t("emptyTask", {taskType: env.t({{::task.type}})})
```

**Comment by crookedneighbor** [comment]
Looking again, and I guess you don't have access to the task object at this point, so ignore my comment about specifying the task type.

Edit: Though, I think you could use `list.type`.  I don't have the ability to work on habit code right now, so I can't check at the moment.

File: website/views/shared/tasks/lists.jade [hunk 1]
```diff
@@ -27,7 +27,9 @@ script(id='templates/habitrpg-tasks.html', type="text/ng-template")
             textarea(rows='6', task-focus='list.bulk && list.focus', ng-model='list.newTask', placeholder='{{list.placeHolderBulk}}', ng-if='list.bulk', ui-keydown='{"meta-enter ctrl-enter":"addTask(obj[list.type+\'s\'],list)"}', required)
             input(type='text', task-focus='!list.bulk && list.focus', ng-model='list.newTask', placeholder='{{list.placeHolder}}', ng-if='!list.bulk', required)
             button(type='submit', ng-disabled='new{{list.type}}form.$invalid')
-              span.glyphicon.glyphicon-plus
+             div.empty-task-notification( ng-show='new{{list.type}}form.$invalid', tooltip=env.t("emptyTask") )
```
Comment (crookedneighbor) [inline website/views/shared/tasks/lists.jade:30]: I think the tooltip should be attached to the button above this, and it should just have conditional text, like this:

```
button(type='submit', ng-disabled='new{{list.type}}form.$invalid', tooltip={{list.newTask == '' ? env.t("emptyTask")) : ''}})
```

Again, doing this from memory, so I may have syntax errors. If it works, this should remove the need for the extra class and surrounding div. It should also position the tooltip better, so it doesn't overlap on the button as much.

**Comment by crookedneighbor** [comment]
@Alys I learned this recently. Adding `?w=1` to the PR url will ignore white space on [Github](https://github.com/blog/967-github-secrets).

**Comment by TheHollidayInn** [comment]
@crookedneighbor Thanks for the suggestion! Unfortunately, since the button is disabled, it does not fire a mouse over event, thus you cannot use the tooltip on it while disabled.

**Comment by TheHollidayInn** [comment]
The first two comments here talk about the disabled buttons: https://github.com/angular-ui/bootstrap/issues/1025 .

**Comment by TheHollidayInn** [comment]
Also, I was using list.type in my Jade code from the first comment, but @Alys recommended a different text in her first comment under my original post.

**Comment by crookedneighbor** [comment]
@TheHollidayInn Regarding `list.type`, the difference is, you were using the untranslated version. (I assume, never saw the original code)

**Comment by crookedneighbor** [comment]
Regarding the other comments, I'll have to look into it later when I can work on the code directly and try a few things, which may take a few weeks, unfortunately. For now, this definitely works.

**Comment by Alys** [comment]
Tested. It works. We can do enhancements later if needed. Thank you! Tier++

Note that the higher tiers require more effort than the lower ones, so if you make any small changes from now on, you might find that they won't grant you an extra tier immediately, but they will be noted as partial credit towards a tier. There's a bit of information about that at [Contributor Rewards](http://habitrpg.wikia.com/wiki/Contributor_Rewards) but do ask if you have questions.

@blade, thanks for the information about ?w=1. Useful!

**Comment by TheHollidayInn** [comment]
@Alys, Thank you! I'll be sure to put more effort in.

Also, I will take a look at your recommendations later today, @crookedneighbor. 

Thanks for all the help you two!

**Pull request merged**